### PR TITLE
Complex with stmt

### PR DESCRIPTION
**Pull request by markww**
This pull request makes the pfff python parser able to handle complex with statements (ie, those that take multiple with_items). This changes the Python With AST node and the generic OtherStmtWithStmt AST node to take a list of with_items and a list of with expressions, respectively.

### CODE REVIEW

File: lang_python/parsing/parser_python.mly [hunk 1]
```diff
@@ -496,8 +496,16 @@ excepthandler:
   | EXCEPT test COMMA test COLON suite { ExceptHandler (Some $2, Some (expr_store $4), $6) }
 
 with_stmt:
-  | WITH test         COLON suite { With ($2, None, $4) }
-  | WITH test AS expr COLON suite { With ($2, Some $4, $6) }
+  | WITH with_list COLON suite { With ($2, $4) }
```
Comment (aryx) [inline lang_python/parsing/parser_python.mly]: Here, you could just process the with_list list and generate 
With (elem1 (With elem2 (With elem3 ...) until the end of the list which you would have $4
Basically something like   fold_right (fun acc e -> With e acc) $2 $4

File: lang_python/parsing/parser_python.mly [hunk 2]
```diff
@@ -496,8 +496,16 @@ excepthandler:
...
+
+with_item:
+  | test            { WithItem ($1, None) }
```
Comment (aryx) [inline lang_python/parsing/parser_python.mly]: Here no need for a special WithItem, just return the pair ($1, None)

File: lang_python/parsing/ast_python.ml [hunk 3]
```diff
@@ -253,7 +255,7 @@ type stmt =
   | While of expr (* test *) * stmt list (* body *) * stmt list (* orelse *)
   | If of expr (* test *) * stmt list (* body *) * stmt list (* orelse *)
   (* https://docs.python.org/2.5/whatsnew/pep-343.html *)
-  | With of expr (* context_expr *) * expr option (* optional_vars *) * stmt list (* body *)
+  | With of with_item list (* context items *) * stmt list (* body *)
```
Comment (aryx) [inline lang_python/parsing/ast_python.ml]: no need to change With. You can just desugar With X as A, Y as B into 2 With, With X as A (With Y as B (...))

File: tests/python/parsing/with_stmt.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+def with_stmt_test():
```
Comment (aryx) [inline tests/python/parsing/with_stmt.py]: good test.

**Comment by markww** [comment]
I'm abandoning this branch in favor of the much more concise complex-with-stmt-2, for which I have created a separate PR.

**Pull request closed without merging**